Library/Homebrew/os/mac/cache_store.rb

```diff
@@ -0,0 +1,207 @@
+require "dbm"
+require "json"
+
+#
+# `DatabaseCache` is a class acting as an interface to a persistent storage
+# mechanism residing in the `HOMEBREW_CACHE`
+#
+class DatabaseCache
+  # Name of the database cache file located at <HOMEBREW_CACHE>/<name>.db
+  #
+  # @return [String]
+  attr_accessor :name
+
+  # Instantiates new `DatabaseCache` object
+  #
+  # @param  [String] name
+  # @return [nil]
+  def initialize(name)
+    @name = name
+  end
+
+  # Memoized `DBM` database object with on-disk database located in the
+  # `HOMEBREW_CACHE`
+  #
+  # @return [DBM] db
+  def db
+    @db ||= DBM.open("#{HOMEBREW_CACHE}/#{name}", 0666, DBM::WRCREAT)
```

Comment:
What's `0666` here? Could be good to document what both of these arguments mean, maybe by using variable names.